Misc fixes

mikechristie opened the pull request:
The following is just some misc patches.

Commit:
Limit log file size to 5MB.

Signed-off-by: Mike Christie <mchristi@redhat.com>

Commit:
Only allow num and alpha chars for image name

Chars like "@" and "/" are not allowable chars for image names. I think
the current regex expression was trying to limit the allowable chars to
numbers, chars, dashes and make sure that there is a "." to separate the
pool and image.

However, we are currently allowing any char like "@#$/" etc.

We seem to need a "^" and "$" to make sure everything between the
beginnging and end of the string is the type listed in the expression
and fail if not. Without them we seem to only check if the image name
has a char, number or dash and as long as it does we return success.

Note: I have no idea what I am doing with regex. I had to google this so
hopefully someone can confirm this works for the reason I think it does.

Signed-off-by: Mike Christie <mchristi@redhat.com>

dillaman commented:
jenkins flake8

dillaman commented on ceph_iscsi_config/lun.py:
```diff
@@ -736,6 +738,113 @@ def remove_dev_from_lio(self):
+        :return: (str) either 'ok' or an error description
+        """
+
+        mode_vars = {"create"      : ['pool', 'image', 'size', 'count'],
```
> ./ceph_iscsi_config/lun.py:748:35: E203 whitespace before ':'
> ./ceph_iscsi_config/lun.py:749:35: E203 whitespace before ':'
> ./ceph_iscsi_config/lun.py:750:35: E203 whitespace before ':'
> ./ceph_iscsi_config/lun.py:751:35: E203 whitespace before ':'

I just merged the tox fixes that I had been sitting on to avoid downstream conflicts, so this probably just needs to be re-copied from ```ceph-iscsi-cli```

dillaman commented on ceph_iscsi_config/lun.py:
```diff
@@ -736,6 +738,113 @@ def remove_dev_from_lio(self):
+        config_obj = Config(logger)
+        if config_obj.error:
+            return "Ceph iSCSI config object not available"
+        config = config_obj.config
```
Does this still work w/ ```ceph-ansible``` on an initial deployment? Just wondering if the ```gateway.conf``` has already been inserted into the pool by this point.

mikechristie replied:
Yes, it works. The LUN() call which is done a little after the valid_disk call in the ceph-ansible patch:

https://github.com/mikechristie/ceph-ansible/commit/b9a4db8b4a18a543c0645821294bfaf54024e1a0

does a Config() too.

Let me actually repush this patch, because for the rbd-target-api valid_disk calls, the caller has a fresh config object. Also when we do LUN() calls we have a config object already too so there is not need to call Config() so many times in a row for this type of case.

mikechristie replied:
Note: I did an explicit test with a fresh config and it does work as expected. The new ceph settings like non standard cluster names will get used ok.

dillaman reviewed (commented):
BTW -- nice regex comment in your commit message

mikechristie commented:
I rebased the patches to the current master.

One additional change to the patches.

I had the caller pass in the config object to valid_disk(). For gwcli this is slightly faster because originally we were doing get_config which did a local http request which just returned the local config object. In my original patches I was doing Config which did a ceph request to get a new copy. In the new patches I just pass in the local config directly.

For ceph-ansible, there is no perf improvement right now.  I just had the caller do Config() and pass that to valid_disk.

Note.

Above I mentioned I also wanted to have the caller to pass in its config object to valid_disk and then also LUN() so LUN() does not have to do a extra Config() call, but I am hitting some issues during failures, so I will do this in another patchset.

dillaman commented:
@mikechristie See the flake8 failures

Commit:
Move valid_disk from gwcli to ceph-iscsi-config

ceph-ansible igw needs to do the same type of checks, so move this to
the config package.

Signed-off-by: Mike Christie <mchristi@redhat.com>

mikechristie commented:
Ran tox locally and they should all be fixed now.

dillaman reviewed (approved):
lgtm
